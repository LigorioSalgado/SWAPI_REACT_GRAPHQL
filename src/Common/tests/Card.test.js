import React from 'react';
import {shallow} from 'enzyme';

import {Card} from '../Card';


describe("<Card/>",() => {

	it("Should render correctly",() => {
		const component = shallow(<Card/>)
		expect(component).toMatchSnapshot();
	})

	it("Should render props", () => {
		const component = shallow(<Card name={"prueba"}>
			<ul>
				<li>Prueba</li>
			</ul>
		</Card>)

		expect(component.find('li')).toHaveLength(1)

	})

	it("Should props name be equal", () => {
		const component = shallow(<Card name={"prueba"}></Card>)

		expect(component.find('.card-title').text()).toBe("prueba")

	})

})