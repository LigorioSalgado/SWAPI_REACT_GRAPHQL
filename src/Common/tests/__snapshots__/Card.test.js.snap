// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> Should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card-body"
      >
        <h5
          className="card-title"
        />
      </div>,
      "className": "card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5
            className="card-title"
          />,
          undefined,
        ],
        "className": "card-body",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "card-title",
          },
          "ref": null,
          "rendered": null,
          "type": "h5",
        },
        undefined,
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card-body"
        >
          <h5
            className="card-title"
          />
        </div>,
        "className": "card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h5
              className="card-title"
            />,
            undefined,
          ],
          "className": "card-body",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "card-title",
            },
            "ref": null,
            "rendered": null,
            "type": "h5",
          },
          undefined,
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
